mne/channels/channels.py

```diff
@@ -987,6 +989,118 @@ def add_channels(self, add_list, force_update_info=False):
+            instance of Raw::
+
+                method = lambda data: np.median(data, axis=0)
+
+            Defaults to ``mean``.
+
+        Returns
+        -------
+        inst : instance of Raw, Epochs, or Evoked
+            A new instance.
+        """
+        from ..io import BaseRaw, RawArray
+        from ..epochs import BaseEpochs
+        from ..evoked import Evoked
+        from .. import EpochsArray, EvokedArray
+
+        # Check to see if data is preloaded
+        _check_preload(self, 'Combining channels')
+
+        ch_axis = len(self._data.shape) - 2  # idx of the channel axis
+        ch_idx = _picks_to_idx(self.info, None)
+        ch_types = _get_channel_types(self.info)
+        group_by = deepcopy(group_by)
+
+        # Convert string values of ``method`` into callables
+        if isinstance(method, str):
+            method_dict = {key: partial(getattr(np, key), axis=ch_axis)
+                           for key in ('mean', 'median', 'std')}
+            try:
+                method = method_dict[method]
+            except KeyError:
+                raise ValueError('"method" must be a callable, or one of '
+                                 '"mean", "median", or "std"; got {}'
+                                 ''.format(method))
+
+        # Check correctness of combinations
+        for this_group, these_picks in group_by.items():
+            # Check if channel indices are out of bounds
+            if not all(idx in ch_idx for idx in these_picks):
+                raise ValueError('Some channel indices are out of bounds.')
+            # Check if heterogeneous sensor type combinations
+            this_ch_type = np.array(ch_types)[these_picks]
+            if len(set(this_ch_type)) > 1:
+                types = ', '.join(set(this_ch_type))
+                raise ValueError('Cannot combine sensors of different types; '
+                                 '"{}" contains types {}.'
+                                 ''.format(this_group, types))
+            #  Check if combining less than 2 channel
+            if len(these_picks) < 2:
+                raise ValueError('Less than 2 channels in group "{}"; cannot '
+                                 'combine by method "{}".'
+                                 ''.format(this_group, method))
+            # If all good create more detailed dict with channel type
+            group_by[this_group] = dict(picks=these_picks,
+                                        ch_type=this_ch_type[0])
+
+        try:
+            # Extract stim channels if any
+            inst = self.copy().pick_types(meg=False, stim=True)
+            # Create clean info to avoid inconsistencies
+            inst.info = create_info(sfreq=self.info['sfreq'],
+                                    ch_names=inst.info['ch_names'],
+                                    ch_types=_get_channel_types(inst.info))
+        except ValueError:
+            inst = None
```

Comment:
can you explain the rationale here?  I can imagine cases where keeping stim channels would be desirable, and other cases where the presence of stim channels would be an unexpected surprise...  seems like something the user should be able to control.

Reply:
Good point on giving the user control on this, I can implement a `keep_stim` boolean argument. Should it defaults to something? Maybe to `False` to avoid unexpected behaviour?

Reply:
+1 for `keep_stim` with default `False`